backend/run.ts

```diff
@@ -14,11 +15,12 @@ export type Inject = {
 
 async function actualRun(appInfo: AppInfo, options: Options, inject: Inject): Promise<void> {
   const { injectFrontend, injectSim, getIndexHtml } = inject;
+  options.basePort = await detectPort(options.basePort);
```

Comment:
Before this line `options.basePort` is either `7000` (the default) or the value provided by `--port` and `detectPort()` then tries that port or finds the next available in sequence.